### PR TITLE
FOGL-3233 Add source configuration item and boost thread library as

### PR DESCRIPTION
**Pull request by MarkRiddoch**
required by latest freeopcua library

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -114,6 +114,9 @@ if (NOT OPCUASERVER)
 endif()
 target_link_libraries(${PROJECT_NAME} ${OPCUASERVER} ${OPCUACORE} ${OPCUAPROTOCOL})
 
+find_package(Boost 1.53.0 COMPONENTS thread REQUIRED)
+target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
```
Comment (ashish-jabble) [inline CMakeLists.txt:118]: If its required with latest freeopcua library. Then same fix applies to `foglamp-south-opcua`?

Comment (MarkRiddoch) [reply CMakeLists.txt:118]: It is related to the opcuaserver library that is only used in the north. After checking the opcua south plugin with the latest freeopcuv there is no problem.

**Review by ashish-jabble — CHANGES_REQUESTED** [review]
Now I see another issue
```
Aug 29 16:23:45 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'op1' to start at 2019-08-29 16:23:46.320034
Aug 29 16:23:45 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Enabled Schedule 'op1/e135a061-8be4-45a0-aa8e-c46a2ab63acd' process 'north_c'
Aug 29 16:23:45 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'op1' process 'north_c' task 7de8471b-f11a-406f-81e7-086b3a6a19a6 pid 15728, 2 running tasks#012['tasks/north_c', '--port=38133', '--address=127.0.0.1', '--name=op1']
Aug 29 16:23:46 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'stats collection' process 'stats collector' task 335143de-b363-4f14-9cda-4779a4bb76bb pid 15742, 3 running tasks#012['tasks/statistics', '--port=38133', '--address=127.0.0.1', '--name=stats collection']
Aug 29 16:23:46 nerd-034 FogLAMP Storage[15580]: ERROR: SQLite3 storage plugin raising error: Not all updates within transaction succeeded
Aug 29 16:23:46 nerd-034 op1[15729]: ERROR: Update table: Not all updates within transaction succeeded
Aug 29 16:23:46 nerd-034 FogLAMP Storage[15580]: ERROR: SQLite3 storage plugin raising error: Not all updates within transaction succeeded
Aug 29 16:23:46 nerd-034 op1[15729]: ERROR: Update table: Not all updates within transaction succeeded
Aug 29 16:23:46 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process terminated: Schedule 'stats collection' process 'stats collector' task 335143de-b363-4f14-9cda-4779a4bb76bb pid 15742 exit 0, 2 running tasks#012['tasks/statistics']
Aug 29 16:23:46 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'stats collection' to start at 2019-08-29 16:24:01.320034
Aug 29 16:24:01 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'stats collection' process 'stats collector' task 8de9d745-a023-4433-ba47-82f8676dd8a5 pid 15795, 3 running tasks#012['tasks/statistics', '--port=38133', '--address=127.0.0.1', '--name=stats collection']
Aug 29 16:24:01 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process terminated: Schedule 'stats collection' process 'stats collector' task 8de9d745-a023-4433-ba47-82f8676dd8a5 pid 15795 exit 0, 2 running tasks#012['tasks/statistics']
Aug 29 16:24:01 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'stats collection' to start at 2019-08-29 16:24:16.320034
Aug 29 16:24:16 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'stats collection' process 'stats collector' task 36e59042-2a84-46b7-a345-3c81323a68cd pid 15879, 3 running tasks#012['tasks/statistics', '--port=38133', '--address=127.0.0.1', '--name=stats collection']
Aug 29 16:24:16 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process terminated: Schedule 'stats collection' process 'stats collector' task 36e59042-2a84-46b7-a345-3c81323a68cd pid 15879 exit 0, 2 running tasks#012['tasks/statistics']
Aug 29 16:24:16 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'stats collection' to start at 2019-08-29 16:24:31.320034
Aug 29 16:24:31 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'stats collection' process 'stats collector' task 4942d75e-dc5d-49ce-856c-637d8a60f280 pid 15928, 3 running tasks#012['tasks/statistics', '--port=38133', '--address=127.0.0.1', '--name=stats collection']
Aug 29 16:24:31 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process terminated: Schedule 'stats collection' process 'stats collector' task 4942d75e-dc5d-49ce-856c-637d8a60f280 pid 15928 exit 0, 2 running tasks#012['tasks/statistics']
Aug 29 16:24:31 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'stats collection' to start at 2019-08-29 16:24:46.320034
Aug 29 16:24:45 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process terminated: Schedule 'op1' process 'north_c' task 7de8471b-f11a-406f-81e7-086b3a6a19a6 pid 15728 exit 0, 1 running tasks#012['tasks/north_c']
Aug 29 16:24:45 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'op1' to start at 2019-08-29 16:24:46.320034
Aug 29 16:24:46 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'op1' process 'north_c' task 237efc1c-fbc7-442d-8ee4-ded0baacf57c pid 15977, 2 running tasks#012['tasks/north_c', '--port=38133', '--address=127.0.0.1', '--name=op1']
Aug 29 16:24:46 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'stats collection' process 'stats collector' task ec6f2e16-34ce-4a74-a7ce-05b46d781c70 pid 15980, 3 running tasks#012['tasks/statistics', '--port=38133', '--address=127.0.0.1', '--name=stats collection']
Aug 29 16:24:46 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process terminated: Schedule 'stats collection' process 'stats collector' task ec6f2e16-34ce-4a74-a7ce-05b46d781c70 pid 15980 exit 0, 2 running tasks#012['tasks/statistics']
Aug 29 16:24:46 nerd-034 FogLAMP[15572] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'stats collection' to start at 2019-08-29 16:25:01.320034
```

**DB results**

```
sqlite> select count(*) from readings;
count(*)
261


sqlite> select * from streams;
id|description|properties|object_stream|object_block|object_filter|active_window|active|last_object|ts
1|op1|{}|{}|{}|{}|{}|t|252|2019-08-29 16:34:26.056


sqlite> select * from statistics;
key|description|value|previous_value|ts
READINGS|Readings received by FogLAMP|261|261|2019-08-29 16:33:35.901
BUFFERED|Readings currently in the FogLAMP buffer|0|0|2019-08-29 16:33:35.901
UNSENT|Readings filtered out in the send process|0|0|2019-08-29 16:33:35.901
PURGED|Readings removed from the buffer by the purge process|0|0|2019-08-29 16:33:35.901
UNSNPURGED|Readings that were purged from the buffer before being sent|0|0|2019-08-29 16:33:35.901
DISCARDED|Readings discarded by the South Service before being  placed in the buffer. This may be due to an error in the readings themselves.|0|0|2019-08-29 16:33:35.901
RANDOMWALK|Readings received from asset randomwalk|261|261|2019-08-29 16:34:02.119
Readings Sent|Readings Sent North|249633405|249633405|2019-08-29 16:34:26.594
op1|op1|249633405|249633405|2019-08-29 16:34:26.636
```

**Comment by MarkRiddoch** [comment]
> Now I see another issue
> 
> ```
> Aug 29 16:23:46 nerd-034 FogLAMP Storage[15580]: ERROR: SQLite3 storage plugin raising error: Not all updates within transaction succeeded
> Aug 29 16:23:46 nerd-034 op1[15729]: ERROR: Update table: Not all updates within transaction succeeded
> Aug 29 16:23:46 nerd-034 FogLAMP Storage[15580]: ERROR: SQLite3 storage plugin raising error: Not all updates within transaction succeeded
> Aug 29 16:23:46 nerd-034 op1[15729]: ERROR: Update table: Not all updates within transaction succeeded

This is not a problem, it is expected behaviour as the code dies to update the streams table first, catches the error and then does an insert. This has been raised previously under Jira FOGL-2593

The stats value is probably the result of old values from a previous run.

**Review by ashish-jabble — APPROVED** [review]

**Comment by ashish-jabble** [comment]
@MarkRiddoch yeah problem is not a plugin specific, Sending process first tries updation, if that fails, it assumes the key doesn't exist in DB table and then sends an insert with same values.
If it is acceptable at the moment then its fine. But I believe we need to fix 2593, as incorrect count does not looks good and more confused.

**Pull request merged**